tests/linalg_test.py

```diff
@@ -68,6 +68,21 @@ def _axis_for_ndim(ndim: int) -> Iterator[None | int | tuple[int, ...]]:
     yield (-1, 0, 1)
 
 
+def _random_invertible(rng, shape, dtype):
+    """
+    Generate a random invertible matrix was specified shape and dtype
+    """
+    invertible = False
+    while not invertible:
+      a = rng(shape, dtype)
+      try:
+        np.linalg.inv(a)
+        invertible = True
+      except np.linalg.LinAlgError:
+        pass
+    return [a]
```

Comment:
Suggestion for a slightly more compact definition – also, please replace the other use of this code with the helper function.

(note in particular: I think `return a` rather than `return [a]` makes more sense in this context)
```suggestion
def _random_invertible(rng, shape, dtype):
    """Generate a random invertible matrix was specified shape and dtype"""
    while True:
      a = rng(shape, dtype)
      try:
        np.linalg.inv(a)
      except np.linalg.LinAlgError:
        pass
      else:
        return a
```